newIDE/app/src/ResourcesList/ResourceUtils.js

```diff
@@ -48,12 +49,42 @@ export const getLocalResourceFullPath = (
   return resourcePath;
 };
 
-export const resourceHasValidPath = (
+export const isResourcePathinProjectFolder = (
+  project: gdProject,
+  resourcePath: string
+) => {
+  const projectPath = path.dirname(project.getProjectFile());
+  return resourcePath.includes(projectPath);
+};
+
+export const userWantsToUseExternalPath = (
```

Comment:
nitpicking: `userWantsToUseExternalPath` looks like it's something that return true or false, but it's actually doing some more complex stuff (i.e: ask something to the user).
Also, I find it unusual to include "user" in the name of the method. I prefer method starting with a verb that describe what is being done. I would suggest: `confirmResourcePath`

confirm hints at the usage of `confirm` method (so a user input will happen). The name is more generic (you could say too generic), but this means that we can later add more validation here.